### PR TITLE
Add mad support for v_pk_* 16 bit integer

### PR DESCRIPTION
**Pull request by dstutt**

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/reassoc-mul-add-1-to-mad.ll [hunk 1]
```diff
@@ -2999,16 +2993,15 @@ define <2 x i16> @v_mul_284_add_82_v2i16(<2 x i16> %arg) {
...
+; GFX9-NEXT:    v_pk_mad_u16 v0, v0, s4, v1 op_sel_hi:[1,0,0]
 ; GFX9-NEXT:    s_setpc_b64 s[30:31]
 ;
 ; GFX10-LABEL: v_mul_284_add_82_v2i16:
 ; GFX10:       ; %bb.0:
 ; GFX10-NEXT:    s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
-; GFX10-NEXT:    v_pk_mul_lo_u16 v0, 0x11c, v0 op_sel_hi:[0,1]
-; GFX10-NEXT:    v_pk_add_u16 v0, 0x52, v0 op_sel_hi:[0,1]
+; GFX10-NEXT:    s_movk_i32 s4, 0x11c
+; GFX10-NEXT:    v_pk_mad_u16 v0, v0, s4, 0x52 op_sel_hi:[1,0,0]
```
Comment (bfavela) [inline llvm/test/CodeGen/AMDGPU/reassoc-mul-add-1-to-mad.ll:3004]: Strange that the constant didn't get folded here. GFX10 supports two constants on every VALU.

Comment (jayfoad) [reply llvm/test/CodeGen/AMDGPU/reassoc-mul-add-1-to-mad.ll:3004]: You can't encode two different literals in one instruction. 0x52 and 0x11c are both too big to be inline constants.

Comment (bfavela) [reply llvm/test/CodeGen/AMDGPU/reassoc-mul-add-1-to-mad.ll:3004]: 0x52 not decimal 52, right. My apologies!

**Review by arsenm — APPROVED** [review]
I'm surprised this was missed. Also remove the change-id from the description before merging

**Comment by dstutt** [comment]
> I'm surprised this was missed. Also remove the change-id from the description before merging

Thanks for removing it.

**Pull request merged**